src/base/encryption.c

```diff
@@ -227,15 +228,18 @@ crypt_decrypt_printable (const char *crypt, char *decrypt, int maxlen)
     {
       return -1;
     }
+  outlen = inlen;
 #elif defined (HAVE_LIBGCRYPT)
-  if (gcry_cipher_decrypt (cipher_Hd, decrypt, maxlen, crypt, inlen))
+  char *dest = NULL;
+  int ec = crypt_default_decrypt (NULL, crypt, inlen, crypt_Key, (int) sizeof crypt_Key, &dest, &outlen, DES_ECB);
```

Comment:
1. For the function **crypt_default_decrypt ()**, I can't find any reference to it, probably it may not need anymore in version 10+.
2. Since **HAVE_RPC_DES_CRYPT_H** is declared, the codes you commented as '// OPENSSL' (final #else directive), seems to have not been compiled. If it was not defined,
    * I think '**inlen**' undefined error seems to occur.
    * When 'src' is not a string encrypted by aes_default_encrypt, **crypt_default_decrypt** () will returns NULL as 'dest' pointer with return value NO_ERROR. Next time, I'm not sure the result of next statements (memcpy (decrypt, NULL, outlen)), and db_private_free_and_init ().

I'm not sure, maybe I didn't understand your intention fully.

Reply:
1. I've added it because I saw that crypt_encrypt_printable seemed to be used and maybe it would be nice to have it. I've deleted it now if you also agree with removing it.

2. Yes.. I seem to have made a couple of mistakes there. Fixed them.

I tried to keep the order of `#ifdefine`s to match what was in the previous code.

Reply:
I agree to the deletion of the **crypt_decrypt_printable ()** function. I will review the rest of the code a bit more. Thank you for your quick response!